### DISCUSSION TITLE
No longer able to publish Unity 2022.3.11f1 Mac builds via MacOS-12 remote-agent

### DISCUSSION THREAD

**Discussion by eeease** [actions/virtual-environments#10339] [General questions]
We use a MacOS agent to create MacOSX Unity builds of our app.  It has worked up until recently, when we upgraded to Unity 2022.3.11f1.

I can't figure out why the step is failing and why PowerShell is exiting with code '1'.  The failure skips the proceeding publishing steps.

We use `macOS-latest`, which is currently `macOS-12`.  I have tried `macOS-13` which also fails in the same way.

Specifically, our Unity Build step fails, but what's bizarre is that the logs show the build was a success:
`Build Finished, Result: Success.
`
Batchmode is also quitting successfully, it seems.

Here's the task we're using on our windows-2019 agent AND the macos-12 agent:
>       - task: PowerShell@2
        displayName: "⚠️ Unity: Build"
        inputs:
          continueOnError: true
          retryCountOnTaskFailure: 3
          showWarnings: true
          targetType: inline
          runScriptInSeparateScope: true
          script: >-
            & "$Env:EDITOR_PATH${{ parameters.unityexecutable }} -quit -batchmode -nographics -projectPath "$(projectDir)"
            -executeMethod Editor.BuildScript.BuildMacDevelopment
            -outputDir "$Env:BUILD_ARTIFACTSTAGINGDIRECTORY\" -buildTarget ${{ parameters.buildtarget }} | Out-Default

I have been staring at our Windows Build step vs our Mac Build and there are some slight differences.  Can anyone shed some light on what these `info: Microsoft.Hosting.Lifetime` logs are (see attached)?

![image-20240724-135539](https://github.com/user-attachments/assets/f4a825de-d833-4131-b0b4-5b9f4a412b87)

**Comment by erik-bershel** [comment]
There are reported issues for Unity builds in GitHub environment. We found one platform related issue and working on the elimination of it. But I am not sure that it is your case. Take a look here, also: https://github.com/actions/runner-images/issues/10219.

**Reply by eeease** [reply]
Thanks!  I'm subscribed to 10219 so I'm keeping an eye on developments there.
We're currently using Azure Pipelines hosted runners but I'm interested in seeing if others are having similar issues.
